### DISCUSSION TITLE
Export ALL users list

### DISCUSSION THREAD

**Discussion by ibousaid** [appwrite/appwrite#7340] [General]
Hello, How to export all users with a query on name and email ? I have tried the code below but I cannot export more than few users and I have all details, I couldn't select only emails for instance. Thank you.

**const sdk = require('node-appwrite');

// Init SDK
const client = new sdk.Client();

const users = new sdk.Users(client);

client
  .setEndpoint('https://appwrite.project.com/v1') // Your API Endpoint
  .setProject('.....') // Your project ID
  .setKey(
    '......' ); // Your secret API key

const promise = users.list();

promise.then(
  function (response) {
    console.log(response);
    console.log("-----");
  },
  function (error) {
    console.log(error);
  }
);`**

**Comment by eldadfux** [comment]
You need to use pagination to iterate over the list or increase the query limit.
https://appwrite.io/docs/products/databases/pagination
